### PR TITLE
fix(issue-priority): Explicitly override priority for issue platform

### PR DESCRIPTION
**Pull request by snigdhas**

### CODE REVIEW

File: src/sentry/issues/ingest.py [hunk 1]
```diff
@@ -105,7 +105,7 @@ def _create_issue_kwargs(
         "type": occurrence.type.type_id,
         "first_release": release,
         "data": materialize_metadata(occurrence, event),
-        "priority": occurrence.initial_issue_priority,
+        "priority": occurrence.initial_issue_priority or occurrence.type.default_priority,
```
Comment (snigdhas) [inline src/sentry/issues/ingest.py:108]: This is technically superfluous, but we call `save_issue_occurrence` in lots of tests which skips setting the default priority. Worth considering if we should disallow that if the intended pipeline for occurrences is via `_process_message`

File: src/sentry/event_manager.py [hunk 2]
```diff
@@ -1878,7 +1878,10 @@ def _create_group(project: Project, event: Event, **kwargs: Any) -> Group:
     group_data["metadata"].update(severity)
 
     if features.has("projects:issue-priority", project, actor=None):
-        priority = _get_priority_for_group(severity, kwargs)
+        priority = kwargs.get("priority", None)
```
Comment (snigdhas) [inline src/sentry/event_manager.py:1881]: Ensures we don't override the priority with log level logic if it's already provided

**Review by wedamija — APPROVED** [review]

File: src/sentry/event_manager.py [hunk 3]
```diff
@@ -1878,7 +1878,10 @@ def _create_group(project: Project, event: Event, **group_creation_kwargs: Any)
     group_data["metadata"].update(severity)
 
     if features.has("projects:issue-priority", project, actor=None):
-        priority = _get_priority_for_group(severity, group_creation_kwargs)
+        priority = group_creation_kwargs.get("priority", None)
+        if priority is not None:
```
Comment (wedamija) [inline src/sentry/event_manager.py:1882]: Mislead you here, meant `is None` 😅

Comment (snigdhas) [reply src/sentry/event_manager.py]: my brain wasn't working either lol

**Pull request merged**